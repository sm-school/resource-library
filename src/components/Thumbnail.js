import PropTypes from 'prop-types';
import React from 'react';

class Thumbnail extends React.Component {
	render(){
		const altText = "Thumbnail image for " + this.props.siteUrl;
		return (
			<div class="thumbnail">
				<img src={this.props.imageDataUrl} alt={altText} />
			</div>
		);
	}
};

Thumbnail.propTypes = {
	imgDataUrl: PropTypes.string,
	siteUrl: PropTypes.string,
};

export default Thumbnail;
