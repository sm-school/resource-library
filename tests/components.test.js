import React from 'react';
import renderer from 'react-test-renderer';

import Thumbnail from '../src/components/Thumbnail';

describe('Thumbnail', () => {
	it('matches the snapshot', () => {
		const tree = renderer.create(
			<Thumbnail
				// not actually an image - doesn't matter here
				imgDataUrl = 'data:image/jpeg;base64,dGVzdCBzdHJpbmc='
				siteUrl = 'nytimes.com'
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
});