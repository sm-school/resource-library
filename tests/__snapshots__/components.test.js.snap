// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail matches the snapshot 1`] = `
<div
  class="thumbnail"
>
  <img
    alt="Thumbnail image for nytimes.com"
    src={undefined}
  />
</div>
`;
